aten/src/ATen/record_function.h

```diff
@@ -368,18 +370,18 @@ class TORCH_API RecordFunctionCallback {
     return scopes_[(size_t)sc];
   }
 
-  inline const std::function<std::unique_ptr<ObserverContext>(const RecordFunction&)>& start() const {
+  inline StartCallback start() const {
```

Comment:
Why do we want a copy here?
```suggestion
  inline const StartCallback& start() const {
```

Reply:
because function pointers are cheap to copy, like `int` or `void*`

Reply:
`std::function` is not just a function pointer, is it?

Reply:
No, but this diff changes StartCallback to be a function pointer above.